api/v1beta1/azuremanagedmachinepool_types.go

```diff
@@ -556,6 +556,11 @@ type AzureManagedMachinePoolSpec struct {
 	// Immutable.
 	// +optional
 	SubnetName *string `json:"subnetName,omitempty"`
+
+	// EnableFIPS indicates whether FIPS is enabled on the node pool.
+	// Immutable.
```

Comment:
In order to make this immutable, you're going to have to add a block similar to [this](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/b07d34e72a9ff577c77342b7eb5d6c610c71335c/api/v1beta1/azuremanagedmachinepool_webhook.go#L170-L175) in `azuremanagedmachinepool_webhook.go`.

Reply:
Thanks for pointing that out @Jont828 . Added.